From deaff8bf47b3fbaa5dc17e0c555524869593282c Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Wed, 9 Mar 2011 09:55:56 +0800
Subject: [PATCH 121/502] ENGR00140389 Lightsensor hal: Add int_lt_lux value
 check.

Set int_lt_lux to 0 if it is negative.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 libsensors/LightSensor.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libsensors/LightSensor.cpp b/libsensors/LightSensor.cpp
index dabc40c..201b16d 100755
--- a/libsensors/LightSensor.cpp
+++ b/libsensors/LightSensor.cpp
@@ -127,6 +127,10 @@ int LightSensor::setIntLux()
     lux = atoi(buf);
     int_ht_lux = lux + mThresholdLux;
     int_lt_lux = lux - mThresholdLux;
+
+    if (int_lt_lux < 0)
+	    int_lt_lux = 0;
+
     DEBUG("Current light is %d lux\n", lux);
 
     /* Set low lux and high interrupt lux for polling */
-- 
1.8.0

